keras_nlp/models/bert.py

```diff
@@ -0,0 +1,253 @@
+        super(Bert, self).__init__(
+            inputs={
+                "input_ids": token_id_input,
+                "segment_ids": segment_id_input,
+                "input_mask": input_mask,
+            },
+            # TODO(jbischof): Consider list output
+            outputs={
+                "sequence_output": sequence_output,
+                "pooled_output": pooled_output,
+            },
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.inner_activation_fn = inner_activation_fn
+        self.initializer_fn = initializer_fn
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.num_segments = num_segments
+        self.intermediate_dim = intermediate_dim
+        self.inner_activation = keras.activations.get(inner_activation)
+        self.initializer_range = initializer_range
+        self.dropout = dropout
+
+    def get_embedding_table(self):
+        return self.get_layer(TOKEN_EMBEDDING_LAYER_NAME).embeddings
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocab_size": self.vocab_size,
+                "hidden_size": self.hidden_size,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "num_segments": self.num_segments,
+                "intermediate_dim": self.intermediate_dim,
+                "inner_activation": keras.activations.serialize(
+                    self.inner_activation
+                ),
+                "dropout": self.dropout,
+                "initializer_range": self.initializer_range,
+            }
+        )
+        return config
+
+
+class BertClassifier(keras.Model):
+    """
+    Classifier model with BertEncoder.
+
+    Args:
+        encoder: A `Bert` Model to encode inputs.
+        num_classes: Number of classes to predict.
+        kernel_initializer: Initializer for the `kernel` weights matrix.
+        bias_initializer: Initializer for the bias vector.
+    """
+
+    def __init__(
+        self,
+        encoder,
+        num_classes,
+        kernel_initializer="glorot_uniform",
+        bias_initializer="zeros",
+        **kwargs,
+    ):
+
+        super().__init__(**kwargs)
+        self.encoder = encoder
+        self.num_classes = num_classes
+        self._logit_layer = keras.layers.Dense(
+            num_classes,
+            kernel_initializer=kernel_initializer,
+            bias_initializer=bias_initializer,
+            name="logits",
+        )
+
+    def call(self, inputs):
+        # Ignore the sequence output, use the pooled output.
+        pooled_output = self.encoder(inputs)["pooled_output"]
+        return self._logit_layer(pooled_output)
+
+
+def BertBase():
```

Comment:
Should we be mirror a kwargs through? For things like `trainable`, `dynamic` or `name`? (generic arguments supported by layers and models)